WordPress/Classes/Networking/ReaderTopicServiceRemote.m

```diff
@@ -174,9 +168,24 @@ - (NSString *)sanitizeTopicNameForAPI:(NSString *)topicName
         topicName = [topicName stringByReplacingOccurrencesOfString:@"--" withString:@"-"];
     }
 
+    topicName = [topicName stringByAddingPercentEscapesUsingEncoding:NSUTF8StringEncoding];
+
     return topicName;
 }
 
+- (NSArray *)normalizeMenuTopicsList:(NSArray *)rawTopics subscribed:(BOOL)subscribed recommended:(BOOL)recommended
+{
+    NSMutableArray *topics = [NSMutableArray array];
+    for (NSDictionary *topicDict in rawTopics) {
+        // Failsafe
+        if (![topicDict isKindOfClass:[NSDictionary class]]) {
+            continue;
+        }
+        [topics addObject:[self normalizeMenuTopicDictionary:topicDict subscribed:subscribed recommended:recommended]];
+    }
+    return [topics copy]; // Return immutable array.
```

Comment:
> wouldn't it create copies for the objects inside the array as well?

Nah. `copy` is a convenience method for `copyWithZone:` and performs a shallow copy of the array.  We could also call `[[NSArray alloc] initWithArray:]` but I prefer this as its shorter.